routes/topics.js

```diff
@@ -16,6 +16,7 @@ router.get('/follow', isAuth, topicsController.getFollowTopic);
 router.get('/latest', validate(TopicValidation.latestPost), isAuth, topicsController.getLatestPost);
 router.put('/follow', isAuth, topicsController.putFollowTopic);
 router.put('/follow-v2', isAuth, topicsController.followTopicV2);
+router.post('/check-name', validate(TopicValidation.checkName), isAuth, topicsController.checkName);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new route `/check-name` is a `POST` request, but it seems like it should be a `GET` request. The purpose of this endpoint is to retrieve information (i.e., whether a name is available), not to create or update resources in the database. According to RESTful API design principles, `GET` requests are used for retrieving data and `POST` requests are used for creating new resources.

```diff
- router.post('/check-name', validate(TopicValidation.checkName), isAuth, topicsController.checkName);
+ router.get('/check-name', validate(TopicValidation.checkName), isAuth, topicsController.checkName);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->